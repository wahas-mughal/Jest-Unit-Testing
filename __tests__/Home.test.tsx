import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import HomeScreen from '../src/Home';
import { describe, expect, it } from '@jest/globals';

describe('Home Screen Snapshot', () => {
  it('Home Screen Snapshot', () => {
    const snap = render(<HomeScreen />).toJSON();
    expect(snap).toMatchSnapshot();
  })
  it('Home Counter Testing', () => {
    const {getByText, getByTestId} = render(<HomeScreen/>);

    //check the initial state of count
    expect(getByTestId('counter-text').props.children).toBe(0);

    //press button simulation
    fireEvent.press(getByText('Increment'));

    //after increment the value should be 1
    expect(getByTestId('counter-text').props.children).toBe(1)
  })
});
