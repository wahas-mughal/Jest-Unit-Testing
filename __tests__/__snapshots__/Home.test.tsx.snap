// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Snapshot Home Screen Snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    testID="counter-text"
  >
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment
      </Text>
    </View>
  </View>
</View>
`;
